WT-8941 Ignore local_retention race in test_tiered10

keitharnoldsmith opened the pull request.

sueloverso reviewed (commented):
This change looks good. I think `test_tiered09.py` has the same issue. Please fix it too.

keitharnoldsmith commented:
Good thinking Sue.  I've added similar checks in `test_tiered09`. I also added a remove of the local file from the second table in that test, since it looks like the code meant to do that, but left out the remove.

sueloverso reviewed (approved):
LGTM thanks.
